$MirOS: ports/net/djbdns/patches/patch-src_dnsq_c,v 1.1.201.1 2005/05/14 23:21:03 tg Exp $
--- src/dnsq.c.orig	Sun Feb 11 21:11:23 2001
+++ src/dnsq.c	Tue Dec 13 23:48:06 2005
@@ -1,3 +1,4 @@
+#include <unistd.h>
 #include "uint16.h"
 #include "strerr.h"
 #include "buffer.h"
@@ -10,6 +11,9 @@
 #include "printpacket.h"
 #include "parsetype.h"
 #include "dns.h"
+#ifdef DJB_V6ONLY
+#include "ip6.h"
+#endif /* DJB_V6ONLY */
 
 #define FATAL "dnsq: fatal: "
 
@@ -24,14 +28,22 @@ void oops(void)
 
 static struct dns_transmit tx;
 
+#ifndef DJB_V6ONLY
 int resolve(char *q,char qtype[2],char servers[64])
+#else
+int resolve(char *q,char qtype[2],char servers[256])
+#endif /* DJB_V6ONLY */
 {
   struct taia stamp;
   struct taia deadline;
   iopause_fd x[1];
   int r;
 
+#ifndef DJB_V6ONLY
   if (dns_transmit_start(&tx,servers,0,q,qtype,"\0\0\0\0") == -1) return -1;
+#else
+  if (dns_transmit_start(&tx,servers,0,q,qtype,V6any) == -1) return -1;
+#endif /* DJB_V6ONLY */
 
   for (;;) {
     taia_now(&stamp);
@@ -47,7 +59,11 @@ int resolve(char *q,char qtype[2],char s
   return 0;
 }
 
+#ifndef DJB_V6ONLY
 char servers[64];
+#else
+char servers[256];
+#endif /* DJB_V6ONLY */
 static stralloc ip;
 static stralloc fqdn;
 
@@ -73,9 +89,15 @@ int main(int argc,char **argv)
 
   if (!*++argv) usage();
   if (!stralloc_copys(&out,*argv)) oops();
+#ifndef DJB_V6ONLY
   if (dns_ip4_qualify(&ip,&fqdn,&out) == -1) oops();
   if (ip.len >= 64) ip.len = 64;
   byte_zero(servers,64);
+#else
+  if (dns_ip6_qualify(&ip,&fqdn,&out) == -1) oops();
+  if (ip.len >= 256) ip.len = 256;
+  byte_zero(servers,256);
+#endif /* DJB_V6ONLY */
   byte_copy(servers,ip.len,ip.s);
 
   if (!stralloc_copys(&out,"")) oops();
